c++/src/vector.cc

```diff
@@ -0,0 +1,100 @@
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <initializer_list>
+#include <cassert>
+
+template <class T>
+class Vector {
+private:
+  unsigned int length = 0;
+  unsigned int capacity = 8;
+  T* dataStore = nullptr;
+
+public:
+  Vector() : dataStore(new T[this->capacity]{0}) {}
+
+  Vector(std::initializer_list<T> list) 
+  : dataStore(new T[this->capacity]{0}) {
+    for (const T &item : list) {
+      this->push(item);
+    }
+  }
+
+  ~Vector() {
+    delete[] this->dataStore;
+  }
+
+
+  void push(T val) {
+    if (length >= capacity) {
+      std::cout << "Vector capacity exceeded. ";
+      std::exit(1);
+    }
+
+    this->dataStore[this->length] = val;
+    this->length++;
```

Comment:
This is perfectly fine. Another way you can do it is to reduce those two lines into one.

``` cpp
this->dataStore[this->length++] = val;
```

It's not clearly better. I think I would keep what you have.